src/robot/libraries/BuiltIn.py

```diff
@@ -1428,6 +1428,21 @@ def set_variable(self, *values):
         else:
             return list(values)
 
+    @run_keyword_variant(resolve=0)
+    def set_local_variable(self, name, *values):
+        """ Makes a variable available everywhere within the local scope.
+
+        Variables set with this keyword are available within the
+        local scope of the currently executed test case or in the local scope
+        of the keyword in which they are defined. For example, if you set a
+        variable in a user keyword, it is available only in that keyword. Other
+        test cases or keywords will not see variables set with this keyword.
+        """
```

Comment:
1. It would be good to mention, probably with an example, that this keyword is equivalent to a normal variable assignment based on a keyword return value.
2. Related to the above, it would be good to mention that the use case is to be able to use this dynamically with `Run Keyword If` and by external test libraries that want to set local variables. Were there some other important use case?
3. A note that this is new in RF 3.2 is a must. See other keywords for examples.
4. There's an unnecessary space after the opening `"""`.